sphinx_immaterial/python_domain_fixes.py

```diff
@@ -545,6 +545,15 @@ def cross_link_single_parameter(
                 new_param_nodes.append(new_param_node)
             param_node.parent[:1] = new_param_nodes
 
+    func_options = apidoc_formatting.get_object_description_options(
+        env, "py", "function"
+    )
+    if (
+        not func_options["cross_link_parameter_descriptions"]
+        or not param_options["cross_link_parameter_descriptions"]
+    ):
+        return
+
```

Comment:
It might make more sense to just skip the call to `_cross_link_parameters` altogether.  I think the only difference is that as you have it now, even if cross linking is disabled, we still call `_add_parameter_links_to_signature`.

Also, I would say that this should just be checking the object description options for the actual object type being defined here (which may be e.g. `py:method` rather than `py:function`), and should not check the object description options for `py:parameter`.

Reply:
1. I wanted to minimize the changes while #99 is open.
2. I think the synopses are closely tied to the cross-linking mechanism. This made it hard for me to discern where would a good place for this.

At first, I had used `self.objtype` to get the options in  https://github.com/jbms/sphinx-immaterial/blob/1062bc88e22bbb731b7cd437e358ef63fd4b22d0/sphinx_immaterial/python_domain_fixes.py#L659-L660 But that didn't cover the `py:parameter` obj type (which for our docs is solely triggered by the autodoc demo).

Reply:
To me it seems more natural that the relevant object type would be the function/method object type (or in the case of C++, any object type).  `py:parameter` does not itself have parameters.  If the user wants to disable cross linking for all python function/method types, they can just use `"py:.*"` as the pattern.

I don't think there should be significant conflicts with #99, since that doesn't make too many changes to `python_domain_fixes.py`.  If there are any conflicts, I can fix them in my PR after this is submitted.

Reply:
If I do insert an condition that prevents `_cross_link_parameter()` being called, then would the docs about the `python_qualify_parameter_ids` config need to mention this option as an additional circumstance?

I'm ok with excluding `py:parameter` objects, but it does feel like the option name implies that they are affected.

Reply:
I actually removed `python_qualify_parameter_ids` in #99, and replaced it with a different mechanism, so I think we can ignore its documentation.

I assumed the intent of the `cross_link_parameter_descriptions` option is to basically disable the special handling of parameters by this theme --- we no longer create `py:parameter` "objects" corresponding to parameters, they don't show up in search results or in the TOC.  But maybe that isn't the behavior you are looking for --- maybe you want to still create the `py:parameter` objects, but just don't link them to the signature.

Reply:
> we no longer create py:parameter "objects" corresponding to parameters

Is this related to #99 changes also? 

>  But maybe that isn't the behavior you are looking for 

I only want to allow the prevention of linking of parameters to descriptions. I honestly don't care if we create a `py:parameter` node/object or not (its an implementation detail - I'm focused on user experience).

As long as any other config options aren't negatively affected by this option, then I have no problem preventing the `_cross_link_parameter()` being called.

Reply:
The main question is --- do you want the parameters to show up in the TOC and search results (as their own entity, rather than just text within the object description), or not?

Reply:
I like the ability to cross-reference a specific parameter, but I'm not concerned with search results or ToC (especially if the description fields are going into the ToC). I could forego the cross-ref-ing ability since that has been the norm for so long.

To be honest, I've been disabling the parameters inclusion in the ToC for any of my projects that use this theme.

Reply:
Perhaps there should be some finer-grained options rather than just this one option to disable the generation of `parameter` objects entirely.  E.g.:

object_description_options:
- `generate_parameter_objects: bool = True`  --- enables generation of parameter objects from `:param xx:` fields in the object description; these parameter objects can be cross-referenced (with synopsis shown depending on `generate_synopsis`), appear in search results, and can be shown in the TOC (depending on `include_in_toc`).
- `cross_link_signature_parameters: bool = True` --- cross-links parameters defined in signatures to the parameter objects defined by `generate_parameter_objects`.  Has no effect if `generate_parameter_objects` is `False`.
- `nitpick_parameter_names: bool = True` --- emits a warning if a parameter documented with `:param:` can't be matched to a parameter defined by a signature.  Has no effect if either `cross_link_signature_parameters` or `generate_parameter_objects` is `False`.

Reply:
I'm just working what is currently there. I'm ok with closing this PR and continue waiting for #99 to be merged, then re-addressing #112 with the improved code-base.

For now, I have to use this patch for developing the clang-autodoc approach because some of my test code (which was taken from breathe's example cases) will trigger the problem stated in #112 .